analyzer/impl/remote.c

```diff
@@ -552,7 +552,8 @@ suscan_remote_read(
     SU_TRYCATCH(poll(fds, 2, timeout_ms) != -1, return -1);
 
     if (fds[1].revents & POLLIN) {
-      (void) read(cancelfd, &cancel_byte, 1);
+      int ignore_result __attribute__((unused));
```

Comment:
I'm worried this only works in GCC and Clang (even if these are the only ones we use for now), how about something like:

```c
#ifdef __GNUC__ /* Or even check for version? */
#  define IGNORE_RESULT(type, expr)              \
  do {                                           \
    type ignored_val__ __attribute__((unused));  \
    ignored_val__ = expr;                        \
  } while (0)   
#else
#  define IGNORE_RESULT(type, expr) (void) expr
#endif /* __GNUC__ */

/* ... */

IGNORE_RESULT(size_t, read(cancelf, &cancel_byte, 1));
```

Reply:
it is pity, there is no common way for it. (I liked the (void) )
Do you have a idea where to put this macro? Or create new header?

Reply:
> it is pity, there is no common way for it. (I liked the (void) )
> Do you have a idea where to put this macro? Or create new header?

Indeed it is. But these are the compilers we have to deal with :(

Anyways, I believe this should be fixed in sigutils level (mainly because that's the root dependency of all projects). I think sigutils' `<util/util.h>` could do the job, and it is implicitly included by almost all header files in the project.

Reply:
updated :)